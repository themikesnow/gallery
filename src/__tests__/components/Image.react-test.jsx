import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { expect as expectChai } from 'chai';
import sinon from 'sinon';

import Image from '../../components/Gallery/Image.react';

const image = {
  src: 'DummyPath',
};

describe('<Image />', () => {
  it('Match Snapshot', () => {    
    const component = renderer.create(<Image image={image} isActive={false} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('componentWillReceiveProps', () => {
    sinon.spy(Image.prototype, 'componentWillReceiveProps');
    const wrapper = mount(<Image image={image} isActive={false} />);
    expectChai(Image.prototype.componentWillReceiveProps).to.have.property('callCount', 0);
    wrapper.setProps({ image: { src: 'Another DummyPath' }, isActive: true });
    expectChai(Image.prototype.componentWillReceiveProps).to.have.property('callCount', 1);
  });
});
