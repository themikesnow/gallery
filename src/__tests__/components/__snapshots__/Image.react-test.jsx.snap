// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Image /> Match Snapshot 1'] = `
<div
  className="image"
  onClick={[Function]}
>
  <img
    alt="Thumbnail"
    src={undefined}
  />
</div>
`;
